File: src/main/java/ru/femirion/leetcode/easy/ConvertDoublyLinkedListToArrayI.java
package ru.femirion.leetcode.easy;

import ru.femirion.leetcode.utils.ListNode;

import java.util.ArrayList;
import java.util.List;

/**
 * 3263. Convert Doubly Linked List to Array I
 * https://leetcode.com/problems/convert-doubly-linked-list-to-array-i/description/
 */
public class ConvertDoublyLinkedListToArrayI {

    /**
     * Runtime 6 ms Beats 6.87%
     * Memory 45.22 MB Beats 6.53%
     */
    public int[] toArray(ListNode head) {
        List<Integer> list = new ArrayList<>();

        while(head != null) {
            list.add(head.val);
            head = head.next;
        }

        return list.stream().mapToInt(e -> e).toArray();
    }
}

File: src/main/go/easy/convert_doubly_linked_list_to_array_i.go
package easy

/*
*

3263. Convert Doubly Linked List to Array I
https://leetcode.com/problems/convert-doubly-linked-list-to-array-i/description/
Runtime 0 ms Beats 100.00%
Memory 5.12 MB Beats 10.87%
*/
func toArray(head *Node) []int {
	var result = make([]int, 0)
	for head != nil {
		result = append(result, head.Val)
		head = head.Next
	}
	return result
}
